include/zephyr/pm/evtc.h

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This file uses the following short names:
+ *
+ *   uptime ticks     -> uptks
+ *   ticks            -> tks
+ *   latency          -> lat
+ *   latencies        -> lats
+ *   event            -> evt
+ *   event controller -> evtc
+ *   activate         -> act
+ *   request          -> req
+ *   runtime          -> rt
+ *   update           -> upd
+ *   timeout          -> to
+ *   effective        -> eff
+ *
+ * This file uses the following terms:
+ *
+ *   event is active    -> event's max latency is accounted for
+ *   event is effective -> event's max latency is in effect
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_EVTC_H_
+#define ZEPHYR_INCLUDE_PM_EVTC_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+
+struct pm_evt;
+struct pm_evtc;
+
+struct pm_evt {
+	sys_snode_t node;
+	const struct pm_evtc *evtc;
+	uint32_t lat_ns;
+	int64_t uptks;
+};
+
+typedef void (*pm_evtc_request)(const struct device *dev, uint32_t latency_ns);
+
+struct pm_evtc_rt {
+	const struct pm_evtc *const evtc;
+	sys_slist_t evt_list;
+	struct k_spinlock lock;
+	uint32_t req_lat_ns;
+	int64_t req_uptks;
+	struct _timeout upd_to;
+};
+
+struct pm_evtc {
+	const struct device *dev;
+	pm_evtc_request req;
+	const uint32_t *lats_ns;
+	uint8_t lats_ns_size;
```

Comment:
it adds a hole in the structure. Re-order to possibly avoid i.

Reply:
I can re-order it